[ruby] failed to build

Just checking to see if this is really an issue from my installed gcc modifications on x86_64.
Does anyone else have issues rebuilding ruby?

```crew reinstall -s ruby```:

```
compiling ossl_x509crl.c
compiling ../.././ext/psych/yaml/scanner.c
compiling ../.././ext/psych/yaml/writer.c
linking shared-object fiddle.so
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/fiddle'
compiling ossl_x509ext.c
compiling ossl_x509name.c
compiling ossl_x509req.c
compiling ossl_x509revoked.c
compiling ossl_x509store.c
installing default openssl libraries
linking shared-object psych.so
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/psych'
linking shared-object openssl.so
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/openssl'
make[1]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2'
make: *** [uncommon.mk:295: build-ext] Error 2
ruby failed to build: `make V=0 -j4` exited with 2
```

```
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/pty'
make[2]: Entering directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/readline'
compiling readline.c
readline.c: In function 'username_completion_proc_call':
readline.c:79:42: error: 'username_completion_function' undeclared (first use in this function); did you mean 'rl_username_completion_function'?
   79 | # define rl_username_completion_function username_completion_function
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
readline.c:1904:37: note: in expansion of macro 'rl_username_completion_function'
 1904 |                                     rl_username_completion_function);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
readline.c:79:42: note: each undeclared identifier is reported only once for each function it appears in
   79 | # define rl_username_completion_function username_completion_function
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
readline.c:1904:37: note: in expansion of macro 'rl_username_completion_function'
 1904 |                                     rl_username_completion_function);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:250: readline.o] Error 1
make[2]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2/ext/readline'
make[1]: *** [exts.mk:264: ext/readline/all] Error 2
make[1]: Leaving directory '/usr/local/tmp/crew/ruby-2.7.2.tar.xz.dir/ruby-2.7.2'
make: *** [uncommon.mk:295: build-ext] Error 2
```

--- a/packages/ruby.rb
+++ b/packages/ruby.rb
@@ -3,22 +3,22 @@ require 'package'
 class Ruby < Package
   description 'Ruby is a dynamic, open source programming language with a focus on simplicity and productivity.'
   homepage 'https://www.ruby-lang.org/en/'
-  version '2.7.2'
+  version '3.0.0'
   compatibility 'all'
-  source_url 'https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.2.tar.xz'
-  source_sha256 '1b95ab193cc8f5b5e59d2686cb3d5dcf1ddf2a86cb6950e0b4bdaae5040ec0d6'
+  source_url 'https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.0.tar.gz'
+  source_sha256 'a13ed141a1c18eb967aac1e33f4d6ad5f21be1ac543c344e0d6feeee54af8e28'
 
   binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-2.7.2-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-2.7.2-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-2.7.2-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-2.7.2-chromeos-x86_64.tar.xz',
+     aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-3.0.0-chromeos-armv7l.tar.xz',
+      armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-3.0.0-chromeos-armv7l.tar.xz',
+        i686: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-3.0.0-chromeos-i686.tar.xz',
+      x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/ruby-3.0.0-chromeos-x86_64.tar.xz',
   })
   binary_sha256 ({
-    aarch64: 'a435e6bf7965e1a82e8842e5ea66bdd670ec9b627d785bd720d3d2652fc89f6d',
-     armv7l: 'a435e6bf7965e1a82e8842e5ea66bdd670ec9b627d785bd720d3d2652fc89f6d',
-       i686: '81865864d3ba93b6cbd5dc8e1b6cb51bd2ebe854f6c01e282c1b73f379fb7caf',
-     x86_64: '658808516b7a2e58f8102fd131e765aaa79f2a7c906d0330b7e883fbdc12d1a9',
+     aarch64: '81509c5d3585f1a0c3c7d616d9f9a0d0192c6e3276c3ddea23c3905e78e56bfc',
+      armv7l: '81509c5d3585f1a0c3c7d616d9f9a0d0192c6e3276c3ddea23c3905e78e56bfc',
+        i686: '6a624ec30b5b6f32a842ec9f686278ce239d729f9736bf1b5d74ac08a716e0ef',
+      x86_64: '64ab9f4df6ab0185f7f74a0deb6a809d5ba910128f730522f21079062985108f',
   })
 
   depends_on 'readline'
@@ -28,12 +28,20 @@ class Ruby < Package
   # at run-time, system's gmp, openssl, readline and zlibpkg are possible to use
 
   def self.build
-    system "./configure #{CREW_OPTIONS} --enable-shared"
+    # The download from ruby-lang.org doesn't need autoconf run,
+    # but the download from github does.
+    #system "autoconf"
+    system "CFLAGS='-ltinfo -flto' \
+    CXXFLAGS='-flto' \
+    optflags='-flto' \
+    ./configure #{CREW_OPTIONS} \
+    --enable-shared \
+    --disable-fortify-source"
     system 'make'
   end
 
   def self.check
-    #system "TMPDIR=#{CREW_PREFIX}/tmp make check"
+    system "TMPDIR=#{CREW_PREFIX}/tmp make check || true"
   end
 
   def self.install

